File: templates/python/day01a.py
# Advent of Code 2017 - Python template

from common import *

def solve(input):
    return input

class Tests():
    import pytest
    def test_example_1(x): assert solve(13) == 13
    def test_example_2(x): assert solve(get_input('template.input.example')) == '13'
    def test_solve(x): assert solve(get_input('template.input'))

if __name__ == "__main__":
    print(solve(get_input('template.input')))

File: templates/julia/day01a.jl
# Advent of Code 2017 - Julia template

include("common.jl")

function solve(input)
    return input
end

# I haven't worked out how to force all the tests to always
#   run with this approach, so it will stop at the first failed
#   test. If you know how to fix this while still keeping the
#   individual solution files clear of noise, please let me know.

# do_test and do_solve are macros defined in common.jl

@do_test 1337 1337
@do_test getInput("template.input.example") "13"
@do_solve getInput("template.input")

